### DISCUSSION TITLE
How can the server tell if it's running in production/staging, or development mode?

### DISCUSSION THREAD

**Discussion by lukehutch** [serverpod/serverpod#2274] [Q&A] [answered]
I need to add the GCP storage configuration to `server.dart`, but only for production and staging, not for development. How can I check which mode Serverpod is running in, at runtime?

**Comment by lukehutch** [comment, marked as answer]
This is what I came up with:

```dart
  final isProduction = pod.runMode == ServerpodRunMode.production ||
      pod.runMode == ServerpodRunMode.staging;
```

**Reply by SandPod** [reply]
That would be the solution to go for right now. We are looking to move away from these 3 predefined values and allow the developer to define their own run modes.

That issues is tracked here: https://github.com/serverpod/serverpod/issues/1212
